samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -303,6 +303,9 @@ private static void createStreams(ProcessorGraph graph, Map<String, SystemAdmin>
   }
 
   private static int maxPartition(Collection<StreamEdge> edges) {
+    if (edges.size() == 0) {
```

Comment:
>>  Is this a bug or for a test case?

This was a bug that occurs for single node graphs. SAMZA-1172 (PR:#100) already fixed this. I'll just re-base with it.